tests/test_public_api.py

```diff
@@ -14,6 +14,58 @@ def model_setup(backend):
     return model, data, init_pars
 
 
+def test_set_backend_by_name(backend):
+    tb, _ = pyhf.get_backend()
+    pyhf.set_backend(tb.name)
+    assert isinstance(
+        pyhf.tensorlib, getattr(pyhf.tensor, "{0:s}_backend".format(tb.name))
+    )
+
+
+@pytest.mark.parametrize("backend_name", ["numpy", "tensorflow", "pytorch"])
+def test_set_backend_by_string(backend_name):
+    pyhf.set_backend(backend_name)
+    assert isinstance(
+        pyhf.tensorlib, getattr(pyhf.tensor, "{0:s}_backend".format(backend_name))
+    )
+
+
+@pytest.mark.parametrize("backend_name", [b"numpy", b"tensorflow", b"pytorch"])
```

Comment:
how does this work in python2?

Reply:
I'm not sure what part of this would be incompatible with Python 2.7. Can you elaborate?

But:

```
$ docker run --rm -it -v $PWD:/home -w /home python:2.7 /bin/bash
# pip install --upgrade pip setuptools wheel
# pip install .
# python
>>> import pyhf
>>> backend_name = b"numpy"
>>> pyhf.set_backend(backend_name)
>>> assert isinstance(
...         pyhf.tensorlib, getattr(pyhf.tensor, "{0:s}_backend".format(backend_name))
...     )
>>> pyhf.tensorlib
<pyhf.tensor.numpy_backend.numpy_backend object at 0x7f644d6425d0>
>>> getattr(pyhf.tensor, "{0:s}_backend".format(backend_name))
<class 'pyhf.tensor.numpy_backend.numpy_backend'>

```

Reply:
ok, this is fine. Just wasn't sure.